Feat/deepl link and navigation

vlonfat opened the pull request:
# Summary

  1. AndroidManifest.xml (+17 lines modified)

  - Added: HTTPS deep link intent filter for invitations
    - Host: joinme.app
    - Path: /invite 
    - android:autoVerify="true" enabled
  - Added: Custom scheme intent filter for joinme://invite

  2. MainActivity.kt (+213 lines)

  - Added newInvitationToken state variable
  - Added invitation processing logic:
    - invitationToken LaunchedEffect
    - pendingInvitationToken LaunchedEffect (to handle when an user need to login itself)
    - handleGroupJoin() helper function
  - Added support for GROUP, EVENT, and SERIES invitation types

I have already tried to increase line coverage on this file but without result. I don't arrive to mock everything. Especially the part with invitation link.

Atilla13 commented:
@Saifurrhn,

Please take a look at #359. One of our teammates, @BryanDMP, encountered the same issue as @vlonfat and was still able to find a solution.

Also, I am not aware of this newly applied rule. I have consistently followed the guideline introduced by one of our assistants, which was **simply** to maintain an overall line coverage of 80%, as you can briefly see in the mail below:

![image](https://github.com/user-attachments/assets/512c0c35-7597-4785-9256-7587b7151872)

Thank you for your understanding.

Saifurrhn commented:
@Atilla13 

What the assistants said is to to maintain an overall line coverage of 80%, BUT that some files are hard to test, like MainActivity. If I remember correctly, you had the same issue a few weeks ago, and that the assistants said that it was okay.

Now for your point about "I have consistently followed the guideline introduced by one of our assistants": If you really listened what our assistants said, you would have known that we have to resolve sonarcloud issues, which was the main reson about why your PR #400 wasn't being accepted (and not for coverage). So if you tell that you follow all guidelines, you would have known what a sonarcloud issue is (Sprint 8 btw, it's a bit sad).

And how you are reviewing here, it shows that you're still angry at @vlonfat, which is very childish, knowing that you were at fault.

I suggest that it's someone else that reviews this PR, to avoid unnecessary problems.

vlonfat commented:
Hello,
Nothing is stopping you from reviewing the code already, given that there are no issues on SonarCloud and the CI passes. This is a first review, so saying that it prevents a review is only relevant in relation to another PR. You can review the code and ask for line coverage to be increased at the same time, which has always been done because line coverage does not impact code changes.

Firstly, as mentioned before, I have already tried to increase line coverage (unit test/Android test). This is not possible because the mocks are far too complex and, due to the structure of the project, it is not really possible to mock a user who is not logged into the app. Furthermore, it is currently not possible to mock a profile. This is being corrected in another issue. 

A lot of the notification code was refactored because when you created the group invitations, there were two overlapping systems, and I asked you to remove one or the other. You removed the Android pop-up from Android. However, you just disabled the display and there was no backend behind it, which created dead code in mainActivity, which was also in notifications. As a result, this removal simplified the notification system and added mainActivity notifications to the line coverage (which was not tested (10 lines)). -> PR: [https://github.com/SWENT-team09-2025/joinMe/pull/218]

Furthermore, since there is only the mainActivity file, there are no files that can be hidden. For example, here is the line coverage of your notification system (link above):  
<img width="1398" height="728" alt="Capture d’écran 2025-12-09 à 10 42 48" src="https://github.com/user-attachments/assets/b1b1dcdf-7398-40fe-9f1a-91adf2f9b2cf" />

You're asking me to do something that you yourself haven't managed to do. I can retry if you give a solution that I didn't try before.

Other example of mainActivity line coverage hidden:
<img width="1422" height="783" alt="Capture d’écran 2025-12-09 à 11 09 20" src="https://github.com/user-attachments/assets/2b03f6c9-5fe8-4426-832e-4294488c0cd6" />
<img width="1421" height="753" alt="Capture d’écran 2025-12-09 à 11 09 53" src="https://github.com/user-attachments/assets/014c1d5a-76f8-4678-8f7e-57dd41671ac7" />

vlonfat commented on app/src/main/java/com/android/joinme/MainActivity.kt:
```diff
@@ -103,12 +107,74 @@ private suspend fun handleGroupJoin(
+    navigationActions: NavigationActions
+) {
+  try {
+    val invitationRepository = InvitationRepositoryFirestore()
```
Yes good thing ! It is my first repo creation since bootcamp ! My bad !

Saifurrhn reviewed (changes requested):
**Overall**
This PR adds invitation link handling into the app flow, including deep-link parsing, token resolution, and navigation to the correct screens. The feature is well integrated and supports all three invitation types (GROUP, EVENT, SERIES), with proper UI feedback and login fallback. The logic is clear and consistent with the existing navigation flow.

**Changes**

- A little change is needed when you inizialize the firestore at each process (commented in code)

**Conclusion**

The feature is well-implemented and integrates invitation handling cleanly into the app flow. Once the minor improvements are addressed, this PR will be ready to merge.

Also here is a comment given by ChatGPT for testing your code:

> Coverage is not increasing because JaCoCo doesn’t instrument code inside Activities after R8 optimization, and Compose-generated bytecode hides most of the logic from the coverage tool. Even with instrumented tests, coverage metrics do not count UI flows, deep link parsing, or LaunchedEffect execution. This is a known limitation and not related to test quality.

So we will accept this low coverage.

Note: This comment was co-written with AI (Claude).

vlonfat commented:
Thank you, for your review. I fixed the comment by adding a new repository provider, modifying the call in mainActivity and added a file test to cover this new file (100% line coverage). It should be ready to review !
